### PR TITLE
feat: add metametrics to authentication

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## **Description**

Allows us to create a lineage (in our system), but will not affect our existing metrics.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24555?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/controllers/authentication/services.ts [hunk 1]
```diff
@@ -57,6 +57,10 @@ export async function login(
       body: JSON.stringify({
         signature,
         raw_message: rawMessage,
+        metametrics: {
+          metametrics_id: clientMetaMetricsId,
+          agent: 'extension',
+        },
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/authentication/services.ts:63]: @dovydas55 is this valid on the API endpoint?

Comment (Prithpal-Sooriya) [reply app/scripts/controllers/authentication/services.ts:63]: Yes this is valid

File: app/scripts/controllers/metametrics.js [hunk 2]
```diff
@@ -652,6 +652,16 @@ export default class MetaMetricsController {
...
 
+  // Retrieve (or generate if doesn't exist) the client metametrics id
+  getMetaMetricsId() {
+    let { metaMetricsId } = this.state;
+    if (!metaMetricsId) {
+      metaMetricsId = this.generateMetaMetricsId();
+      this.store.updateState({ metaMetricsId });
+    }
+    return metaMetricsId;
+  }
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/metametrics.js:663]: This is the only change to metametrics.

We are not using the server metametrics ID on the clients.

**Review by danjm — APPROVED** [review]
This looks good to me

**Review by matteoscurati — APPROVED** [review]
Ok, LGTM

**Pull request merged**